src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -88,9 +88,34 @@ def create_clone(self, pool, source_volname, source_subvolume, snapname):
 
         subvol_path = os.path.join(self.base_path, str(uuid.uuid4()).encode('utf-8'))
         try:
+            # source subvolume and snapshot attrs are used to create clone subvolume
```

Comment:
I'm thinking if it would help to move the bulk of the attribute applying stuff to subvolume base or fs_util source. The helper can accept attributes to apply so as to be reused by cloner.

Reply:
Addressed.